internal/bminventory/inventory.go

```diff
@@ -213,6 +223,19 @@ func NewBareMetalInventory(
 	}
 }
 
+func (b *bareMetalInventory) updatePullSecret(pullSecret string, log logrus.FieldLogger) string {
+
+	res := pullSecret
+	if b.Config.RhQaRegCred != "" {
+		ps, err := validations.AddRHRegPullSecret(pullSecret, b.Config.RhQaRegCred)
+		if err != nil {
+			log.Infof("Failed to add RH QA Credentials to Pull Secret: %s", err.Error())
+		}
+		res = ps
+	}
+	return res
```

Comment:
Seems that in case of  an error you just log it and return what you got from validations.AddRHRegPullSecret.
If that is the desired behavior?
If so I'd update the code a bit
```suggestion
	if b.Config.RhQaRegCred != "" {
		pullSecret, err := validations.AddRHRegPullSecret(pullSecret, b.Config.RhQaRegCred)
		if err != nil {
			log.Infof("Failed to add RH QA Credentials to Pull Secret: %s", err.Error())
		}
	}
	return pullSecret
```

Reply:
I thought that for this part continuing the flow maybe makes more sense.
What do you think? Fail the create/update of PS?

Reply:
Failing the flow in case of error.